pkg/bootstrap/init.go

```diff
@@ -41,86 +40,36 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/manager"
 )
 
+var (
+	CsResources = []string{"csNamespace", "odlmSubscription", "extraResources", "csOperandRegistry", "csOperandConfig"}
+)
+
 // InitResources initialize resources at the bootstrap of operator
 func InitResources(mgr manager.Manager) error {
 	client := mgr.GetClient()
 	reader := mgr.GetAPIReader()
-
-	resourcesDir := os.Getenv("RESOURCES_DIR")
-
-	// create namespace
-	klog.Info("create ibm-common-services namespace")
-	namespace, err := ioutil.ReadFile(filepath.Join(resourcesDir, "namespace.yaml"))
-	if err != nil {
+	if err := createOrUpdateResourcesFromAnnotation(client, reader); err != nil {
 		return err
 	}
 
-	objects, err := yamlToObjects(namespace)
-	if err != nil {
-		return err
-	}
-	if err := createObject(objects[0], client); err != nil {
-		return err
-	}
+	return nil
+}
 
-	klog.Info("check existing ODLM operator")
-	err = deleteExistingODLM(client)
-	if err != nil {
-		return err
-	}
-	// install operator
-	klog.Info("install ODLM operator")
-	subscription, err := ioutil.ReadFile(filepath.Join(resourcesDir, "odlm-subscription.yaml"))
+func getDeployment(reader client.Reader) (*appsv1.Deployment, error) {
+	deploy := &appsv1.Deployment{}
+	deployName, err := k8sutil.GetOperatorName()
 	if err != nil {
-		return err
+		return nil, fmt.Errorf("could not find the operator name: %v", err)
 	}
-
-	if err := createOrUpdateFromYaml(subscription, client, reader); err != nil {
-		return err
-	}
-
-	// create extra yamls
-	klog.Info("create extra yaml resources")
-	if err := createOrUpdateResourcesFromDir(filepath.Join(resourcesDir, "extra"), client, reader); err != nil {
-		return err
-	}
-
-	// create operandConfig and operandRegistry
-	klog.Info("create OperandConfig and OperandRegistry")
-	operandConfig, err := ioutil.ReadFile(filepath.Join(resourcesDir, "cs-operandconfig.yaml"))
+	deployNs, err := k8sutil.GetOperatorNamespace()
 	if err != nil {
-		return err
+		return nil, fmt.Errorf("could not find the operator namespace: %v", err)
 	}
-	operandRegistry, err := ioutil.ReadFile(filepath.Join(resourcesDir, "cs-operandregistry.yaml"))
+	err = reader.Get(context.TODO(), types.NamespacedName{Name: deployName, Namespace: deployNs}, deploy)
 	if err != nil {
-		return err
-	}
-
-	timeout := time.After(300 * time.Second)
```

Comment:
I think we still need to wait for the crd to be ready then deploy config and registry instances.

Reply:
done